build.gradle.kts

```
@@ -219,110 +199,119 @@ tasks.named("wrapper") {
 // General IDE customization                                                                                         //
 ///////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
 
-tasks.register("copyInMissingTemplates", CopyButNeverOverwrite) {
+tasks.register<CopyButNeverOverwrite>("copyInMissingTemplates") {
     description = "Copies in placeholders from the /templates dir to project root if not present yet"
     from(templatesDir)
     into(rootDir)
     include("gradle.properties", "override.cfg")
 }
 
-tasks.register("jmxPassword", CopyButNeverOverwrite) {
+tasks.register<CopyButNeverOverwrite>("jmxPassword") {
     description = "Create config/jmxremote.password from a template."
 
-    setFileMode(0600)  // passwords must be accessible only by owner
+    fileMode = 600  // passwords must be accessible only by owner
```

This is not quite the same? Octal numbers can start with a `0` prefix. File modes in unix-like operating systems are sometimes represented as a triplet of 3 numbers in octal, I think.

---

with 0 it gives a compile errori needed to google as well, in kotlin it is written without the 0.

---

Kotlin [does not support octal literals](https://kotlinlang.org/docs/numbers.html#literal-constants-for-numbers). The equivalent of the octal number `0600` is 384 (decimal) or 0x180 (hexadecimal).

```suggestion
    fileMode = 0x180  // passwords must be accessible only by owner
```

---

@soloturn Let's try something different here (see https://docs.gradle.org/8.3/userguide/working_with_files.html#sec:setting_file_permissions):

```suggestion
    filePermissions { unix("600") } // passwords must be accessible only by owner
```